enhancements/installer/kubevirt-ipi.md

```diff
@@ -0,0 +1,346 @@
+
+         *Note:* In disconnected environment the user will need to provide a local image that the installer
+          can upload to the namespace.
+    - secrets for the Ignition configs of the VMs
+    - 1 bootstrap machine
+    - 3 masters
+    
+    Only on network option 1 - pods network:
+    - Services and routes for DNS and LB
+
+3. Bootstrap
+
+    The bootstrap VM has a huge Ignition config set using terraform as secrets and is visible
+as secrets on the infra OpenShift. KubeVirt boots that VM with that content as ConfigDrive and the 
+bootstraping begins when the `bootkube.service` systemd service starts.
+
+    This process described more thoroughly in the [installer overview document][https://github.com/OpenShift/installer/blob/37b99d8c9a3878bac7e8a94b6b0113fad6ffb77a/docs/user/overview.md#cluster-installation-process]
+
+4. Masters bootstrap
+
+    Master VMs are booting using a stub Ignition config that are waiting early in
+the Ignition service to load their Ignition config from a URL. That URL is the
+ `https://<internal-api-vip>/config/master` which is still not available until
+the **bootstrap** VM is exposing it. It takes few minutes till it does.
+
+    When the MachineConfigServer is available on the bootstrap, the masters pull their Ignition config 
+    and boot up joining the tenant cluster as masters and start scheduling pods.
+
+5. Workers bootstrap
+
+    After the masters and the control plane is up, we will scale the MachineSet to create workers
+    by the machine-api-operator.
+
+
+### Risks and Mitigations
+
+- Network
+
+    - Pods network option
+        -(OCP gap) The ports 22623/22624 that are used by MCS are blocked on the 
+        pods network and prevent from the nodes to pull Ignition config and updates.
+        - (KubeVirt gap) Interface binding - Currently the only supported binding on the pods
+        network is masquerade which means that all nodes are behind NAT, each VM
+        behind the NAT of his own pod.
+        - (OpenShift/KubeVirt gap) Static IP - OpenShift assumes that node's IP addresses are static,
+         but KubeVirt VMs change IP between restarts.
+        
+    - Secondary network option (MULTUS)
+    
+        - With this approach admin of the infra cluster will need to be involved in
+    the creation of each new tenant cluster since NADs need to be created and
+    probably also nmstate will need to be used to create the topology on the hosts.
+    In this proposal, we will assume that admin created the namespace and all network resources 
+    before running the installer, and the created networkName (NAD) will be the input for the installer.  
+
+    
+- Storage
+
+    CSI driver for `KubeVirt` is not available yet.
+
+
+## Design Details
+
+- Namespaces
+
+    For each tenant cluster we will create a namespace with the ClusterID.
+
+    *Open question: should the namespace creation done by the user or by the installer.*
```

Comment:
IPI implies we do everything.  However, some users may have prescribed naming schemes that we should conform to.  I'd recommend we give them the options to specify a name, but we create it.

Reply:
When we moved to the networking option of using Multus, we are not able anymore to create the namespace, since the input of the installer is NAD resource name and the resource must exists in the namespace before we start theinstallation.